src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -971,6 +977,94 @@ private void ReadCompressedTextChunk(PngMetadata metadata, ReadOnlySpan<byte> da
             {
                 metadata.TextData.Add(new PngTextData(name, uncompressed, string.Empty, string.Empty));
             }
+
+            if (name.Equals("Raw profile type exif", StringComparison.OrdinalIgnoreCase))
+            {
+                this.ReadLegacyExifTextChunk(baseMetadata, uncompressed);
+            }
+        }
+
+        /// <summary>
+        /// Reads exif data encoded into a text chunk with the name "raw profile type exif".
+        /// This method was used by ImageMagick, exiftool, exiv2, digiKam, etc, before the
+        /// 2017 update to png that allowed a true exif chunk.
+        /// </summary>
+        /// <param name="metadata">The <see cref="ImageMetadata"/> to store the decoded exif tags into.</param>
+        /// <param name="data">The contents of the "raw profile type exif" text chunk.</param>
+        private void ReadLegacyExifTextChunk(ImageMetadata metadata, string data)
+        {
+            ReadOnlySpan<char> dataSpan = data.AsSpan();
+            dataSpan = dataSpan.TrimStart();
+
+            if (!dataSpan.Slice(0, 4).ToString().Equals("exif", StringComparison.OrdinalIgnoreCase))
+            {
+                // "exif" identifier is missing from the beginning of the text chunk
+                return;
+            }
+
+            // Skip to the data length
+            dataSpan = dataSpan.Slice(4).TrimStart();
+            int dataLengthEnd = dataSpan.IndexOf('\n');
+            int dataLength = int.Parse(dataSpan.Slice(0, dataSpan.IndexOf('\n')).ToString());
+
+            // Skip to the hex-encoded data
+            dataSpan = dataSpan.Slice(dataLengthEnd).Trim();
+            string dataSpanString = dataSpan.ToString().Replace("\n", string.Empty);
+            if (dataSpanString.Length != (dataLength * 2))
+            {
+                // Invalid length
+                return;
+            }
+
+            // Parse the hex-encoded data into the byte array we are going to hand off to ExifProfile
+            byte[] dataBlob = new byte[dataLength - ExifHeader.Length];
+            for (int i = 0; i < dataLength; i++)
+            {
+                byte parsed = Convert.ToByte(dataSpanString.Substring(i * 2, 2), 16);
```

Comment:
Note: I'm not familiar with the algorithm, so maybe this comment may not be applicable.

Couldn't there be used `Encoding.UTF8.GetBytes` (ideally with span-overloads)?
The built-in method is highly optimized, and I doubt that the char-for-char loop will be faster -- code readability should be better with the .NET provided method too.

Or is this done manually with this loop because of the check in L1026?

Reply:
This code looks like it's converting a hexadecimal substring eg `0F`, not an ASCII/utf8 value. The `*.Substring() -> string` allocation is very unfortunate inside a loop though. I don't see span overloads in `Convert.*` methods, wonder if there is a corresponding Span API. If not, implementing our own helper method should not be hard.

Reply:
Correct, it is converting hexadecimal string values to a byte array. The only sort of applicable Convert method here would be https://docs.microsoft.com/en-us/dotnet/api/system.convert.fromhexstring?view=net-6.0, but that is only available with .NET 5+, and it would still require allocating many byte arrays (since we would only be able to parse one line at a time). I'll add a comment along these lines to the code, but the "raw profile type exif" chunk content looks like this:

```

exif
     194
45786966000049492a000800000003000e01020044000000320000001192040001000000
2a000000139202004600000076000000000000004120636f6c6f7266756c2074696c696e
67206f6620626c75652c207265642c2079656c6c6f772c20616e6420677265656e203478
3420706978656c20626c6f636b732e004475706c6963617465642066726f6d206261736e
337030322e706e672c207468656e20696d616765206d65746164617461206d6f64696669
6564207769746820657869763200

```

Like @antonfirsov mentioned, it's probably best to just do this via a helper method that doesn't allocate.

Reply:
I have implemented `HexStringToBytes` that does not allocate. Mine is (IMO) a more readable and safer version of https://stackoverflow.com/a/9995303/871842 that also checks the input data is valid. I'm not sure if @antonfirsov or @gfoidl have something to say about exceptions being used in a hot loop like this. I benchmarked it against the exact code given in the stackoverflow answer above (which has no data validation or exceptions), and there was no difference in performance with 8MB of hex data.

Reply:
Forgot to post these benchmark results in case anyone is curious:

```
|                 Method |     Mean |    Error |   StdDev |
|----------------------- |---------:|---------:|---------:|
| StackOverflowConverter | 75.38 ms | 0.745 ms | 0.582 ms |
|            MyConverter | 71.99 ms | 1.373 ms | 1.737 ms |
|        NativeConverter | 14.39 ms | 0.283 ms | 0.474 ms |
```

NativeConverter is `Convert.FromHexBytes()` available in .NET 5 and newer.

Reply:
You mean [`Convert.FromHexString`](https://source.dot.net/#System.Private.CoreLib/Convert.cs,c9e4fbeaca708991) no?

Reply:
You're right, my mistake. I forgot the reference source was a thing. What's the license for that, and should that implementation just be re-implemented here? Basically it's just using a lookup table instead.

Reply:
MIT so you can reimplement here 😁. I would create a static `HexEncoder` type so we can easily test and benchmark it.

Reply:
@JimBobSquarePants Finally had some more time to look at this. Where should the `HexEncoder` class be created?

Reply:
Stick it under Common/Helpers as an internal class.